spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaKraftBroker.java

```diff
@@ -0,0 +1,593 @@
+	 * @param topic the topic.
+	 */
+	@Override
+	public void consumeFromAnEmbeddedTopic(Consumer<?, ?> consumer, String topic) {
+		consumeFromEmbeddedTopics(consumer, topic);
+	}
+
+	/**
+	 * Subscribe a consumer to one of the embedded topics.
+	 * @param consumer the consumer.
+	 * @param seekToEnd true to seek to the end instead of the beginning.
+	 * @param topic the topic.
+	 * @since 2.8.2
+	 */
+	@Override
+	public void consumeFromAnEmbeddedTopic(Consumer<?, ?> consumer, boolean seekToEnd, String topic) {
+		consumeFromEmbeddedTopics(consumer, seekToEnd, topic);
+	}
+
+	/**
+	 * Subscribe a consumer to one or more of the embedded topics.
+	 * @param consumer the consumer.
+	 * @param topicsToConsume the topics.
+	 * @throws IllegalStateException if you attempt to consume from a topic that is not in
+	 * the list of embedded topics (since 2.3.4).
+	 */
+	@Override
+	public void consumeFromEmbeddedTopics(Consumer<?, ?> consumer, String... topicsToConsume) {
+		consumeFromEmbeddedTopics(consumer, false, topicsToConsume);
+	}
+
+	/**
+	 * Subscribe a consumer to one or more of the embedded topics.
+	 * @param consumer the consumer.
+	 * @param topicsToConsume the topics.
+	 * @param seekToEnd true to seek to the end instead of the beginning.
+	 * @throws IllegalStateException if you attempt to consume from a topic that is not in
+	 * the list of embedded topics.
+	 * @since 2.8.2
+	 */
+	@Override
+	public void consumeFromEmbeddedTopics(Consumer<?, ?> consumer, boolean seekToEnd, String... topicsToConsume) {
+		List<String> notEmbedded = Arrays.stream(topicsToConsume)
+				.filter(topic -> !this.topics.contains(topic))
+				.collect(Collectors.toList());
+		if (notEmbedded.size() > 0) {
+			throw new IllegalStateException("topic(s):'" + notEmbedded + "' are not in embedded topic list");
+		}
+		final AtomicReference<Collection<TopicPartition>> assigned = new AtomicReference<>();
+		consumer.subscribe(Arrays.asList(topicsToConsume), new ConsumerRebalanceListener() {
+
+			@Override
+			public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+			}
+
+			@Override
+			public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+				assigned.set(partitions);
+				LOGGER.debug(() -> "partitions assigned: " + partitions);
+			}
+
+		});
+		int n = 0;
+		while (assigned.get() == null && n++ < 600) { // NOSONAR magic #
+			consumer.poll(Duration.ofMillis(100)); // force assignment NOSONAR magic #
+		}
+		if (assigned.get() != null) {
+			LOGGER.debug(() -> "Partitions assigned "
+					+ assigned.get()
+					+ "; re-seeking to "
+					+ (seekToEnd ? "end; " : "beginning"));
+			if (seekToEnd) {
+				consumer.seekToEnd(assigned.get());
+			}
+			else {
+				consumer.seekToBeginning(assigned.get());
+			}
+		}
+		else {
+			throw new IllegalStateException("Failed to be assigned partitions from the embedded topics");
+		}
+		LOGGER.debug("Subscription Initiated");
+	}
+
+	@Override
+	public void setZkPort(int zkPort) {
```

Comment:
I don't think this method belongs to this class.
So, probably just bite a bullet and remove it from the interface as well?